test/controllers/api/v2/parameters_controller_test.rb

```diff
@@ -21,6 +21,19 @@ def setup
     assert_not_nil assigns(:parameters)
     parameters = ActiveSupport::JSON.decode(@response.body)
     assert_not_empty parameters
+    assert_equal parameters['total'], 1
+  end
+
+  test "should get all parameters of a specific domain" do
+    mydomain = domains(:mydomain)
+    FactoryBot.create(:domain_parameter, :domain => mydomain, :name => 'additional_parameter', :value => '123')
```

Comment:
I was thinking about creating 50
```suggestion
    FactoryBot.create_list(:domain_parameter, 50, :domain => mydomain)
```

This may require you to enhance the factory to generate names and values as a sequence:
```ruby
factory :domain_parameter, :parent => :parameter, :class => DomainParameter do
  type { 'DomainParameter' } 

  sequence(:name) { |n| "parameter#{n}" } 
  sequence(:value) { |n| "value#{n}" } 
end
```

But it may be better to override the `entries_per_page` setting to paginate with 1:
```ruby
with_temporary_settings(entries_per_page: 1) do
  # ...
end
```

And then write the same test except you don't set `per_page` to `all` to assert it paginates.

Reply:
I added a loop in the test to create various parameters. this is pretty easy.